File: app/src/main/java/com/hencoder/hencoderpracticedraw1/practice/Practice8DrawArcView.java
package com.hencoder.hencoderpracticedraw1.practice;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Paint;
import android.graphics.Paint.Style;
import android.graphics.RectF;
import android.support.annotation.Nullable;
import android.util.AttributeSet;
import android.view.View;

public class Practice8DrawArcView extends View {

    public Practice8DrawArcView(Context context) {
        super(context);
    }

    public Practice8DrawArcView(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public Practice8DrawArcView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    Paint mPaint = new Paint(Paint.ANTI_ALIAS_FLAG);
    RectF mRectF = new RectF(200,100,800,500);

    @Override
    protected void onDraw(Canvas canvas) {
        super.onDraw(canvas);

//        练习内容：使用 canvas.drawArc() 方法画弧形和扇形

        mPaint.setStyle(Style.FILL);
        canvas.drawArc(mRectF,-130,110,true,mPaint);// 从-130开始，跨度110

        canvas.drawArc(mRectF,20,120,false,mPaint);

        mPaint.setStyle(Style.STROKE);
        canvas.drawArc(mRectF,180,40,false,mPaint);


    }
}

File: app/src/main/java/com/hencoder/hencoderpracticedraw1/practice/Practice8DrawArcView.kt
package com.hencoder.hencoderpracticedraw1.practice

import android.content.Context
import android.graphics.Canvas
import android.graphics.Paint
import android.graphics.RectF
import android.util.AttributeSet
import android.view.View

/**
 * Created by xh on 2017/10/16.
 */
class Practice8DrawArcView : View {

    private val mPaint = Paint(Paint.ANTI_ALIAS_FLAG)
    private val mRectF = RectF()

    private val mWidth = 350f
    private val mHeight = 250f

    private var mX = 0f
    private var mY = 0f

    constructor(context: Context?) : super(context)
    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs)
    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    override fun onSizeChanged(w: Int, h: Int, oldw: Int, oldh: Int) {
        super.onSizeChanged(w, h, oldw, oldh)

        mX = (w - mWidth) / 2
        mY = (h - mHeight) / 2

    }

    override fun onDraw(canvas: Canvas?) {
        super.onDraw(canvas)

        //        练习内容：使用 canvas.drawArc() 方法画弧形和扇形

        mRectF.set(mX, mY, mX + mWidth, mY + mHeight)

        canvas!!.drawArc(mRectF, -10f, -100f, true, mPaint)
        canvas.drawArc(mRectF, 20f, 140f, false, mPaint)

        mPaint.style = Paint.Style.STROKE
        canvas.drawArc(mRectF, -120f, -60f, false, mPaint)

    }

}